waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -49,6 +58,49 @@ validateExactlyOneAppExists spec =
         GenericValidationError $
           "You have more than one 'app' declaration in your Wasp app. You have " ++ show (length apps) ++ "."
 
+validateWasp :: AppSpec -> [ValidationError]
+validateWasp spec = validateWaspVersion specWaspVersionStr
+  where
+    specWaspVersionStr = Wasp.version $ App.wasp (snd $ getApp spec)
+
+validateWaspVersion :: String -> [ValidationError]
+validateWaspVersion specWaspVersionStr = eitherUnitToErrorList $ do
+  specWaspVersionRange <- parseWaspVersionRange specWaspVersionStr
+  unless
+    ( SV.isVersionInRange currentWaspVersion specWaspVersionRange
+    )
+    $ Left $ incompatibleVersionError currentWaspVersion specWaspVersionRange
+  where
+    parseWaspVersionRange :: String -> Either ValidationError SV.Range
+    parseWaspVersionRange waspVersionRangeStr = do
+      let (_ :: String, _ :: String, _ :: String, waspVersionRangeDigits :: [String]) =
+            waspVersionRangeStr =~ ("\\^([0-9]+).([0-9]+).([0-9]+)$" :: String)
+
+      waspSpecVersion <- case mapM readMaybe waspVersionRangeDigits of
+        Just [major, minor, patch] -> Right $ SV.Version major minor patch
+        __ -> Left $ GenericValidationError "Wasp version should be in the format ^major.minor.patch"
```

Comment:
That would be great! In that case, the `Wasp version should be in the format ^major.minor.patch` would be inaccurate. I was thinking of something like `Invalid Wasp version range format. Check out the documentation for information on supported version range specifiers: <link to docs>`.

Reply:
We could! I wanted to make it simple for the start though -> otherwise we can go and support all of the formats? Not sure what is the best hm. I thought this would be easier to parse. For now it doesn't really matter that they can pin the specific version, we are in beta and the point was just to give them something that works, and I thought `^` is a bit nicer than just fixed version. But we could do both.
TLDR; good with me, but we can also leave that for later and then support everything.

Reply:
Cool. I think I can leave it as it is now, then